circuits/src/withdraw.rs

```diff
@@ -0,0 +1,182 @@
+use std::{collections::HashMap, env::current_dir, time::Instant};
+
+use nova_scotia::{
+    circom::reader::load_r1cs, create_public_params, create_recursive_circuit, FileLocation, F1,
+    G2, S1, S2,
+};
+use nova_snark::{traits::Group, CompressedSNARK};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use ff::PrimeField;
+
+#[derive(Serialize, Deserialize)]
+#[derive(Debug)]
+#[allow(non_snake_case)]
+struct Withdraw {
+    step_in: [String; 1],
+    sender: Vec<String>,
+    recipient: Vec<String>,
+    leaf_coins: Vec<[String; 2]>,
+    pathElements: Vec<Vec<String>>,
+    pathIndices: Vec<Vec<String>>,
+    r: Vec<Vec<String>>,
+    s: Vec<Vec<String>>,
+    msghash: Vec<Vec<String>>,
+    pubkey: Vec<[Vec<String>; 2]>,
+}
+
+pub fn nova(iteration_count: usize) {
+    let root = current_dir().unwrap();
+
+    let circuit_file = root.join("circuits/build/compiled_circuit/withdraw.r1cs");
+    let r1cs = load_r1cs(&FileLocation::PathBuf(circuit_file));
+    let witness_generator_wasm = root.join("circuits/build/compiled_circuit/withdraw_js/withdraw.wasm");
+
+    let withdraw_data: Withdraw = serde_json::from_str(include_str!("../inputs/withdraw.json")).unwrap();
+
+    println!("json: {:?}", withdraw_data);
+
+    let start_public_input = vec![
+        F1::from_str_vartime(&withdraw_data.step_in[0]).unwrap(),
+    ];
+
+    let mut private_inputs = Vec::new();
+
+    for i in 0..iteration_count {
```

Comment:
We can probably use serde to make this a `HashMap`, like I mentioned in the [other review](https://github.com/BlakeMScurr/Altitude/pull/23#discussion_r1234547781).

Reply:
perhaps can be done in a future PR?

Reply:
i took a look at the stackoverflow link and if you're already familiar with how it's done, probably it's better if you make that change in any of your next PR?

Reply:
Yeah true, no need to do now!